jerry-core/api/jerry.c

```diff
@@ -201,23 +192,39 @@ jerry_cleanup (void)
 
   jmem_finalize ();
   jerry_make_api_unavailable ();
-
-  if (JERRY_CONTEXT (user_context_deinit_cb))
-  {
-    JERRY_CONTEXT (user_context_deinit_cb) (JERRY_CONTEXT (user_context_p));
-  }
 } /* jerry_cleanup */
 
 /**
- * Retrieve user context.
+ * Retrieve a context data item, or create a new one.
+ *
+ * @param manager_p pointer to the manager whose context data item should be returned.
  *
- * @return the user-provided context-specific pointer
+ * @return a pointer to the user-provided context-specific data item for the given manager, creating such a pointer if
+ * none was found.
  */
 void *
-jerry_get_user_context (void)
+jerry_get_context_data (const jerry_context_data_manager_t *manager_p)
 {
-  return JERRY_CONTEXT (user_context_p);
-} /* jerry_get_user_context */
+  void *ret = NULL;
+  jerry_context_data_header_t *item_p;
+
+  for (item_p = JERRY_CONTEXT (context_data_p); item_p != NULL; item_p = item_p->next_p)
+  {
+    if (item_p->manager_p == manager_p)
+    {
+      return JERRY_CONTEXT_DATA_HEADER_USER_DATA (item_p);
+    }
+  }
+
+  item_p = jmem_heap_alloc_block (sizeof (jerry_context_data_header_t) + manager_p->bytes_needed);
```

Comment:
@zherczeg Aaaah, I see! You wanted to expose the header to let others do the allocation. Still, at the project level the dev can juggle the relative size of the JerryScript heap wrt. whatever other heap they're using. After all, the total amount of memory on any given device is fixed.

Nevertheless, #1815 is still around so we can go with that one.